### PR TITLE
Lemma 7.4.2, `adjointCarleson_adjoint`, `BoundedCompactSupport`

### PR DESCRIPTION
**Pull request by roos-j**
This proves Lemma 7.4.2 and various prerequisites that will also be useful for other lemmas. 
Main additions are:

In `ForestOperator/AlmostOrthogonality.lean`:
- `adjoint_tree_estimate`: Lemma 7.4.2
- `adjointCarleson_adjoint`: `adjointCarleson` is actually the adjoint of `carlesonOn`
- `BoundedCompactSupport.adjointCarleson`: `adjointCarleson` has bounded compact support if the input function does

In `TileStructure.lean`:
- `AEStronglyMeasurable.carlesonOn`: `carlesonOn` is a.e. strongly measurable
- `BoundedCompactSupport.carlesonOn`: `carlesonOn` has bounded compact support if the input function does

In `Psi.lean`:
- `exists_bound_of_norm_Ks`: existence of uniform bound for `‖Ks x y‖` if `x` is in a bounded set

The new file `BoundedCompactSupport.lean` defines a predicate for functions that are (essentially) bounded, a.e. strongly measurable and have compact support. It will be convenient to use this as an assumption on input functions in most lemmas and to use/expand helper lemmas in this file as needed.

(Edited 12/8/24)

### CODE REVIEW

File: Carleson/ToMathlib/BoundedCompactSupport.lean [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+/-- Bounded compactly supported measurable functions -/
+-- There are various alternative definitions one could use here
+-- For now I used the formulations that were already used throughout `ForestOperator`
+structure BoundedCompactSupport : Prop where
+  bounded : IsBounded (range f)  -- could use a version of essential boundedness instead,
+                                 -- e.g. `∃ M, ∀ᵐ x ∂μ, ‖f x‖ ≤ M`
+  compact_support : HasCompactSupport f -- could use bounded support instead
+  measurable : AEStronglyMeasurable f -- could use `Measurable` instead
+
+-- Why is there no `IsEssBounded` predicate in mathlib?
```
Comment (fpvandoorn) [inline Carleson/ToMathlib/BoundedCompactSupport.lean:48]: We'll discuss this more tomorrow, but this is `memℒp f ⊤ μ`

File: Carleson/ToMathlib/BoundedCompactSupport.lean [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+-/
+
+namespace MeasureTheory
+
+open Bornology Function Set HasCompactSupport
+open scoped ENNReal
+
+-- Can generalize to vector-valued, but for this project scalar-valued should be enough
+variable {X 𝕜} [MeasureSpace X] [RCLike 𝕜] {f : X → 𝕜}
```
Comment (fpvandoorn) [inline Carleson/ToMathlib/BoundedCompactSupport.lean:35]: For Mathlib: 
`X` should be a `MeasurableSpace` with a specific measure `μ`. The codomain of `f` should be any normed space. (See `memℒp` for a similar definition)

File: Carleson/ForestOperator/AlmostOrthogonality.lean [hunk 3]
```diff
@@ -84,17 +86,251 @@ lemma adjoint_tile_support2 (hu : u ∈ t) (hp : p ∈ t u) : adjointCarleson p
     adjoint_tile_support1, indicator_indicator, ← right_eq_inter.mpr]
   exact (ball_subset_ball (by gcongr; norm_num)).trans (t.ball_subset hu hp)
 
+section ToBeMovedToAppropriateLocations
+
+-- mathlib should have this, but I can't find it
```
Comment (fpvandoorn) [inline Carleson/ForestOperator/AlmostOrthogonality.lean:91]: [Set.indicator_mul_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupWithZero/Indicator.html#Set.indicator_mul_left) is almost this, but maybe these versions should indeed also exist.

File: Carleson/ForestOperator/AlmostOrthogonality.lean [hunk 4]
```diff
@@ -84,17 +86,251 @@ lemma adjoint_tile_support2 (hu : u ∈ t) (hp : p ∈ t u) : adjointCarleson p
...
+-- lemma _root_.Set.indicator_eq_mul_indicator_one {ι M:Type*} [MulZeroOneClass M]
+--     (s : Set ι) (f : ι → M) (x : ι) : s.indicator f x = f x * s.indicator 1 x := by
+--   simp only [indicator]; split_ifs <;> simp
+
+lemma _root_.Set.indicator_eq_indicator_one_mul {ι M:Type*} [MulZeroOneClass M]
+    (s : Set ι) (f : ι → M) (x : ι) : s.indicator f x = s.indicator 1 x * f x := by
+  simp only [indicator]; split_ifs <;> simp
+
+lemma _root_.Set.conj_indicator {α 𝕜 : Type*} [RCLike 𝕜] {f : α → 𝕜} (s : Set α) (x : α):
```
Comment (fpvandoorn) [inline Carleson/ForestOperator/AlmostOrthogonality.lean:100]: special case of [Set.indicator_comp_of_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Indicator.html#Set.indicator_comp_of_zero), but perhaps useful to add specifically.

File: Carleson/ForestOperator/AlmostOrthogonality.lean [hunk 5]
```diff
@@ -84,17 +86,251 @@ lemma adjoint_tile_support2 (hu : u ∈ t) (hp : p ∈ t u) : adjointCarleson p
...
+    conj (s.indicator f x) = s.indicator (conj f) x := by
+  simp only [indicator]; split_ifs <;> simp
+
+-- in mathlib?
+theorem _root_.MeasureTheory.integral_const_mul {X : Type*} [MeasurableSpace X] {μ : Measure X}
```
Comment (fpvandoorn) [inline Carleson/ForestOperator/AlmostOrthogonality.lean:105]: [MeasureTheory.integral_mul_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Integral/Bochner.html#MeasureTheory.integral_mul_left)

**Review by fpvandoorn — COMMENTED** [review]
Some comments already. We can discuss more tomorrow.

File: Carleson/ForestOperator/AlmostOrthogonality.lean (hunk 5 above)
Comment (roos-j) [reply Carleson/ForestOperator/AlmostOrthogonality.lean:105]: Great, thanks!

**Comment by roos-j** [comment]
`BoundedCompactSupport` now uses `Memℒp f ⊤` as discussed

**Comment by fpvandoorn** [comment]
This looks great! Thanks!

**Pull request merged**